client.go

```diff
@@ -30,8 +31,9 @@ type Client struct {
 	Net            string            // if "tcp" or "tcp-tls" (DNS over TLS) a TCP query will be initiated, otherwise an UDP one (default is "" for UDP)
 	UDPSize        uint16            // minimum receive buffer for UDP messages
 	TLSConfig      *tls.Config       // TLS connection configuration
-	Timeout        time.Duration     // a cumulative timeout for dial, write and read, defaults to 0 (disabled) - overrides DialTimeout, ReadTimeout and WriteTimeout when non-zero
-	DialTimeout    time.Duration     // net.DialTimeout, defaults to 2 seconds - overridden by Timeout when that value is non-zero
+	Dialer         *net.Dialer       // a net.Dialer used to set local address, timeouts and more
+	Timeout        time.Duration     // a cumulative timeout for dial, write and read, defaults to 0 (disabled) - overrides DialTimeout, ReadTimeout, WriteTimeout when non-zero. Can be overridden with net.Dialer.Timeout (see Client.ExchangeWithDialer and Client.Dialer) or context.Context.Deadline (see the deprecated ExchangeContext)
```

Comment:
Please put some line breaks in this line - I know there is no limit in Go, but this is pushing it.

Reply:
Will do, not a big fan of long lines either

Reply:
`gofmt` keeps squeezing spaces and breaks the alignment of this and the other (original ) lines. Do you want me to keep the alignment?

Reply:
I've respected `gofmt` for now, which squeeze away the alignment. Have a look and let me know if you prefer the old format, I'll have to disable the `gofmt` plugin to get it aligned :D